overrides/default.nix

```diff
@@ -2241,8 +2241,8 @@ lib.composeManyExtensions [
               );
 
               ARROW_HOME = _arrow-cpp;
-              arrowCppVersion = lib.versions.majorMinor _arrow-cpp;
-              pyArrowVersion = lib.versions.majorMinor prev.pyarrow;
+              arrowCppVersion = lib.versions.majorMinor (_arrow-cpp.version or _arrow-cpp);
```

Comment:
I don't understand what this is supposed to do.
The `or` case would pass the derivation to `majorMinor`, not the version string?

Reply:
I am also hitting this error right now. I think this code never worked in the past. Just change both lines to `.version`. Backwards compatibility is not needed.

Reply:
@TyberiusPrime I would really love to see this merged. Can you remove the `or` and just use `.version`? I think this PR would be ready to go then.